lib/modules/showImages.js

```diff
@@ -644,286 +636,366 @@ function domainToModuleName(hostname) {
+		this.updateButton();
+		other.updateButton();
 	}
 
-	function updateParentFromExpandoResize() {
-		function updateParentHeight(basisHeight) {
-			// .expando-button causes a line break
-			const expandoHeight = Array.from(parent.querySelectorAll('.res-expando, .expando-button'))
-				.reduce((a, b) => a + b.getBoundingClientRect().height, 0);
+	buildDuplicate(original) {
+		this.duplicate = true;
+		this.original = original;
+
+		this.button.addEventListener('click', () => {
+			let swap = true;
+
+			const box = original.box;
+			if (box && box.parentElement) {
+				if (original.options.type === 'IFRAME') {
+					// iframe reloads when reattached after being moving in DOM / becoming parentless, so avoid that
+					original.button.scrollIntoView();
+					swap = false;
+				} else {
+					// Let revealImage reinsert the box by making it parentless
+					box.parentElement.removeChild(box);
+				}
+			}
 
-			parent.style.maxHeight = `${basisHeight + expandoHeight}px`;
-		}
+			if (swap) original.swapWith(this);
+			original.expand();
+		});
 
-		const type = $(expandLink).closest('body:not(.comments-page) .expando')[0] || $(expandLink).closest('.thing')[0];
-		const parent = $(expandLink).closest('.md')[0];
+		this.setAsReady();
+	}
 
-		if (parent && type) {
-			const selfMax = parseInt(module.options.selfTextMaxHeight.value, 10);
-			const commentMax = parseInt(module.options.commentMaxHeight.value, 10);
+	buildUnique(options, siteModule) {
+		this.options = options;
+		this.siteModule = siteModule;
 
-			if (type.classList.contains('expando') && selfMax) {
-				expandLink.expandoBox.addEventListener('resize', () => { updateParentHeight(selfMax); });
-			} else if (type.classList.contains('comment') && commentMax) {
-				expandLink.expandoBox.addEventListener('resize', () => { updateParentHeight(commentMax); });
-			}
+		this.open = false;
+
+		this.button.addEventListener('click', e => {
+			this.update(!this.open);
+			SelectedEntry.handleClick(e);
+		}, true);
+
+		if (isPageType('comments')) {
+			// Execute expando collapse procedure when comment is collapsed
+			this.$button
+				.parents('.comment')
+				.find('> .entry .tagline > .expand')
+				.click(::this.collapse);
 		}
-	}
 
-	if (module.options.autoMaxHeight.value) updateParentFromExpandoResize();
-}
+		this.options.autoExpand = this.options.muted ||
+			['IMAGE', 'GALLERY', 'TEXT'].includes(this.options.type);
 
-function revealImage(expandoButton, showHide) {
-	// don't reveal images for invisible buttons (offsetParent is a cheaper way of checking
-	// visibility than jquery's .is(':visible'))
-	if ((!expandoButton) || (!expandoButton.offsetParent)) {
-		return false;
-	}
-	// showhide = false means hide, true means show!
+		let reveal = false;
+		if (module.options.autoExpandSelfText.value && this.thing.isSelfPost() && this.inText) {
+			// Expand the all autoExpand-expandos in additon to the first non-autoExpand expando
+			reveal = reveal || this.options.autoExpand ||
```

Comment:
`reveal = reveal || this.options...` -> `reveal = this.options...`

(it's always false here)